phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java

```diff
@@ -1333,10 +1334,22 @@ public static Long getPageSizeInMs(ReadOnlyProps props) {
         return null;
     }
 
+    private static void setScanAtrributesForChildLinkRepair(Scan scan, PTable table, PhoenixConnection phoenixConnection) {
```

Comment:
You can move this to the ChildLinkScanTask.java file. Thus will run the verification logic only when needed instead of every time there is a query on the CHILD_LINK table.